### PR TITLE
Faster mangle

### PR DESCRIPTION
**Pull request by UplinkCoder**

### CODE REVIEW

File: src/dmd/dmangle.d [hunk 1]
```diff
@@ -1171,6 +1204,7 @@ extern (C++) const(char)* mangleExact(FuncDeclaration fd)
         OutBuffer buf;
         scope Mangler v = new Mangler(&buf);
         v.mangleExact(fd);
+        v.__dtor();
```
Comment (ibuclaw) [inline src/dmd/dmangle.d:1207]: Give it a proper name, rather than calling an automatic name (See https://github.com/dlang/dmd/pull/11154)

**Comment by UplinkCoder** [comment]
So far this turns to use more memory instead of less.
Something is going wrong.

**Comment by ibuclaw** [comment]
> So far this turns to use more memory instead of less.
> Something is going wrong.

Well, for a start, every object new'd in the compiler has just grown by 4-8 bytes.  So whatever saving you're making is now at odds with what you're loosing.

**Comment by UplinkCoder** [comment]
@ibuclaw Oh the serial thing.
that's accounted for.
It's not that bad.

**Comment by wilzbach** [comment]
Can this experiment be closed?

**Comment by UplinkCoder** [comment]
yep

**Pull request closed without merging**